addons/stock_account/models/stock.py

```diff
@@ -87,83 +88,211 @@ def _get_price_unit(self):
             move = self.search([('product_id', '=', self.product_id.id),
                          ('state', '=', 'done'), 
                          ('location_id.usage', '=', 'internal'), 
-                         ('location_dest_id.usage', '!=', 'internal')], order='date desc', limit=1)
+                         ('location_dest_id.usage', '!=', 'internal'),
+                         ('company_id', '=', self.company_id.id)], order='date desc', limit=1)
             if move:
                 return move.price_unit or self.product_id.standard_price
         return self.product_id.standard_price
 
     def _update_future_cumulated_value(self, value):
         self.ensure_one()
         moves = self.search([('state', '=', 'done'), 
-                     ('date', '>',  self.date), 
-                     ('product_id', '=', self.product_id.id)])
+                     ('date', '>=',  self.date),
+                     ('id', '>', self.id), 
+                     ('product_id', '=', self.product_id.id), 
+                     ('company_id', '=', self.company_id.id),
+                     '|', '&', ('location_id.usage', 'in', ('internal', 'transit')), 
+                    ('location_dest_id.usage', 'not in', ('internal', 'transit')), 
+                    '&', ('location_id.usage', 'not in', ('internal', 'transit')), 
+                    ('location_dest_id.usage', 'in', ('internal', 'transit'))])
         for move in moves:
-            move.value += value
+            move.cumulated_value += value
+
+    def _get_in_domain(self):
+        return [('product_id', '=', self.product_id.id), 
+                ('state', '=', 'done'), 
+                ('company_id', '=', self.company_id.id), 
+                ('location_id.usage', 'not in', ('internal', 'transit')), 
+                ('location_dest_id.usage', 'in', ('internal', 'transit'))]
+
+    def _get_out_domain(self):
+        return [('product_id', '=', self.product_id.id), 
+                ('state', '=', 'done'), 
+                ('company_id', '=', self.company_id.id), 
+                ('location_id.usage', 'in', ('internal', 'transit')), 
+                ('location_dest_id.usage', 'not in', ('internal', 'transit'))]
+
+    def _get_all_domain(self):
+        return [('product_id', '=', self.product_id.id), 
+                 ('state', '=', 'done'), 
+                 ('company_id', '=', self.company_id.id),
+                 '|', '&', ('location_id.usage', 'in', ('internal', 'transit')), 
+                    ('location_dest_id.usage', 'not in', ('internal', 'transit')), 
+                    '&', ('location_id.usage', 'not in', ('internal', 'transit')), 
+                    ('location_dest_id.usage', 'in', ('internal', 'transit'))]
+        
+    def _is_in_move(self):
+        return self.location_id.usage not in ('internal', 'transit') and self.location_dest_id.usage in ('internal', 'transit')
```

Comment:
If it is within the company, it should not be considered in/out.  I was thinking about changing it in: not self.location_id.company_id and self.location_dest_id.company_id == current_company.  Otherwise it is not company_id and location transit (the condition becomes more complicated)